nextcord/interactions/message_component.py

```diff
@@ -447,191 +341,62 @@ async def edit_original_message(
-            embeds=embeds,
-            file=file,
-            files=files,
-            view=view,
-            tts=tts,
-            ephemeral=ephemeral,
-            delete_after=delete_after,
-            allowed_mentions=allowed_mentions,
-            flags=flags,
-            suppress_embeds=suppress_embeds,
-        )
+        Fetches the original interaction response message associated with the interaction.
 
-    async def edit(self, *args, **kwargs) -> Optional[Message]:
-        """|coro|
+        If the interaction response was :meth:`MessageComponentInteractionResponse.send_message` then this would
+        return the message that was sent using that response. Otherwise, this would return
+        the message that triggered the interaction.
 
-        This is a shorthand function for helping in editing messages in
-        response to a component or modal submit interaction. If the
-        interaction has not been responded to, :meth:`InteractionResponse.edit_message`
-        is used. If the response :meth:`~InteractionResponse.is_done` then
-        the message is edited via the :attr:`Interaction.message` using
-        :meth:`Message.edit` instead.
+        Repeated calls to this will return a cached value.
 
         Raises
         ------
         HTTPException
-            Editing the message failed.
-        InvalidArgument
-            You specified both ``embed`` and ``embeds`` or ``file`` and ``files``.
-        TypeError
-            An object not of type :class:`File` was passed to ``file`` or ``files``.
-        HTTPException
-            Editing the message failed.
-        InvalidArgument
-            :attr:`Interaction.message` was ``None``, this may occur in a :class:`Thread`
-            or when the interaction is not a component or modal submit interaction.
+            Fetching the original response message failed.
+        ClientException
+            The channel for the message could not be resolved.
 
         Returns
         -------
-        Optional[:class:`Message`]
-            The edited message. If the interaction has not yet been responded to,
-            :meth:`InteractionResponse.edit_message` is used which returns
-            a :class:`Message` or ``None`` corresponding to :attr:`Interaction.message`.
-            Otherwise, the :class:`Message` is returned via :meth:`Message.edit`.
+        InteractionMessage
+            The original interaction response message.
         """
-        if not self.response.is_done():
-            return await self.response.edit_message(*args, **kwargs)
-        if self.message is not None:
-            return await self.message.edit(*args, **kwargs)
-        raise InvalidArgument(
-            "Interaction.message is None, this method can only be used in "
-            "response to a component or modal submit interaction."
+
+        if self._original_message is not None:
+            return self._original_message
+
+        # TODO: fix later to not raise?
```

Comment:
?

Reply:
This is a Todo inserted by someone else - was just copied over. I am guessing it still wants doing at some point

Reply:
--- from old code; Discussed in nextcord support server ---